src/robot/libraries/BuiltIn.py

```diff
@@ -2220,8 +2224,17 @@ def wait_until_keyword_succeeds(self, retry, retry_interval, name, *args):
             if count <= 0:
                 raise ValueError('Retry count %d is not positive.' % count)
             message = '%d time%s' % (count, s(count))
-        retry_interval = timestr_to_secs(retry_interval)
+        match = re.search(r'(strict *: *)', retry_interval)
+        if match:
+            strict_interval = True
+            retry_interval = retry_interval.lstrip(match.group())
+        else:
+            strict_interval = False
+
+        retry_interval = float(timestr_to_secs(retry_interval))
+        wait_time = retry_interval
         while True:
+            execution_time = time.time()
```

Comment:
`start_time` would be a better name than `execution_time`. The latter sounds like it would be the time execution took. This is especially confusing when there's separate `keyword_runtime` variable that is the actual execution time.

`wait_time` could perhaps be renamed to `sleep_time` because that's the exact time used for sleeping but also the former is fine.